custom/icds_reports/static/js/directives/indie-map/indie-map.directive.js

```diff
@@ -356,7 +360,15 @@ function IndieMapController($scope, $location, $filter, storageService, location
 
     vm.attemptToDrillToLocation = function (geography) {
         var location = getLocationNameFromGeography(geography);
-        locationsService.tryToNavigateToLocation(location, location_id);
+        locationsService.getLocationByNameAndParent(location, location_id).then(function (locations) {
+            var location = locations[0];
+            if (!location) {
```

Comment:
Is this check to check if no location is returned from backend after selected a parent.? Then wont `locations` be empty array and not with first element as null.

Reply:
Good point! I think it still works because `[][0]` resolves to `undefined` but that would be a good thing to cleanup in a follow up PR

Reply:
https://github.com/dimagi/commcare-hq/pull/26449/commits/9b4d47de6220252feebcd3e2be2fc09eb46fc407